# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/include/platform/FeaturesD3D.h
# Commit: 754436d8b3a2
# Full Hash: 754436d8b3a2023b90eb247b3a70335e74f2fdf6
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-05-09 09:14:23
# Regressor Bug: 1633628
# File Overlap Count: 3
# Description:
#   Bug 1633628 - Vender: Don't use ClearView if we previously used dual source blending on Intel gen6. r=jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D74137
# ==============================================================================

diff -r f8fd6f0a5102 -r 754436d8b3a2 gfx/angle/checkout/include/platform/FeaturesD3D.h
--- a/gfx/angle/checkout/include/platform/FeaturesD3D.h	Fri May 08 18:39:28 2020 +0000
+++ b/gfx/angle/checkout/include/platform/FeaturesD3D.h	Fri May 08 19:28:53 2020 +0000
@@ -155,6 +155,14 @@
                               "On some Intel drivers, using clear() may not take effect", &members,
                               "https://crbug.com/655534"};
 
+    // On Sandybridge, calling ClearView after using dual source blending causes the hardware to hang.
+    // See: https://bugzilla.mozilla.org/show_bug.cgi?id=1633628
+    Feature emulateClearViewAfterDualSourceBlending = {"emulate_clear_view_after_dual_source_blending",
+                              FeatureCategory::D3DWorkarounds,
+                              "On Sandybridge, calling ClearView after using dual source blending causes "
+                              "the hardware to hang", &members,
+                              "https://bugzilla.mozilla.org/show_bug.cgi?id=1633628"};
+
     // On some Intel drivers, copying from staging storage to constant buffer storage does not
     // seem to work. Work around this by keeping system memory storage as a canonical reference
     // for buffer data.